src/nested_ragged_tensors/ragged_numpy.py

```
@@ -989,124 +1217,172 @@
+                            tensors[new_key] = T[S]
+                case _:
+                    raise TypeError(f"{type(idx)} not supported for {self.__class__.__name__} slicing")
+
+            schema[new_key] = tensors[new_key].dtype
+
+        out = self.__class__(processed_tensors=tensors, schema=schema)
+        if squeeze_dims is not None:
+            i = 0
+            for dim in sorted(squeeze_dims):
+                out = out.squeeze(dim - i)
+                i += 1
```

Ruff

Use `enumerate()` for index variable `i` in `for` loop

(SIM113)